### PR TITLE
Make dashboard plugin tests ready to CI

### PR DESCRIPTION
**Pull request by kandl**
- Update the structure a bit, add 2 plugins for the tests (with latest version for new API(s), and 8.8.0 to test backward compability)
- Add more tests
- Refresh recordings
- Add CI script

JIRA: RAIL-3888

---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

File: common/scripts/ci/run_dashboard_plugin_tests.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+DIR=$(echo $(cd $(dirname "${BASH_SOURCE[0]}") && pwd -P))
+_RUSH="${DIR}/docker_rush.sh"
+_RUSHX="${DIR}/docker_rushx.sh"
+
+$_RUSH install
+$_RUSH build -t dashboard-plugin-tests
+$_RUSHX tools/dashboard-plugin-tests build
```
Comment (no23reason) [inline common/scripts/ci/run_dashboard_plugin_tests.sh:9]: Is this second build needed? It should already be built by the previous command, right? The lowercase `-t` builds the target package as well, as opposed to `-T` that does not build the target package.

File: tools/dashboard-plugin-tests/cypress/integration/filterBar/filterBar.spec.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+// (C) 2019-2022 GoodData Corporation
+import { withTestConfig } from "../../tools/configure";
+import { PLUGIN_LATEST_CONFIG } from "../../support/constants";
+
+// customize.layout().customizeFluidLayout()
```
Comment (no23reason) [inline tools/dashboard-plugin-tests/cypress/integration/filterBar/filterBar.spec.ts:5]: Maybe forgetten code to be removed?

File: tools/dashboard-plugin-tests/plugins/plugin-8.8.0/.gitignore [hunk 3]
```diff
@@ -1,3 +1,2 @@
 .env.secrets
 dist
-node_modules
```
Comment (no23reason) [inline tools/dashboard-plugin-tests/plugins/plugin-8.8.0/.gitignore:3]: Is this intentional? Or do the node_modules get ignored by some other gitignore?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7242/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9092/ : SUCCESS in 17m 16s

File: tools/dashboard-plugin-tests/cypress/integration/filterBar/filterBar.spec.ts (hunk 2 above)
Comment (kandl) [reply tools/dashboard-plugin-tests/cypress/integration/filterBar/filterBar.spec.ts:5]: Oh, I forgot to update the comment, it should point relevant tested API - updated

File: tools/dashboard-plugin-tests/plugins/plugin-8.8.0/.gitignore (hunk 3 above)
Comment (kandl) [reply tools/dashboard-plugin-tests/plugins/plugin-8.8.0/.gitignore:3]: This is interesting, it's basically generated from the plugin cli tool, previous plugin was just removed... But it's likely not necessary, as we have `node_modules/` in gitignore in the root folder of the SDK

File: common/scripts/ci/run_dashboard_plugin_tests.sh (hunk 1 above)
Comment (kandl) [reply common/scripts/ci/run_dashboard_plugin_tests.sh:9]: Removed

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7245/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9095/ : SUCCESS in 17m 16s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2092/ : SUCCESS in 12m 10s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2376/ : SUCCESS in 18s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1195/ : SUCCESS in 15m 47s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9098/ : SUCCESS in 17m 50s

**Pull request merged**